akka-docs/src/main/paradox/stream/operators/Source/maybe.md

```diff
@@ -14,9 +14,35 @@ Materialize a @scala[`Promise[Option[T]]`] @java[`CompletionStage`] that if comp
 
 ## Description
 
-Materialize a @scala[`Promise[Option[T]]`] @java[`CompletionStage`] that if completed with a @scala[`Some[T]`] @java[`Optional`] 
-will emit that *T* and then complete the stream, or if completed with @scala[`None`] @java[`empty Optional`] complete the stream right away.
+Materialize a @scala[`Promise[Option[T]]`] @java[`CompletableFuture`] which controls what element will be emitted by the
```

Comment:
Possibly focus on the source bit of it. Like "Create a source that emits once the materialized promise is completed with a value..."
"This makes it possible to inject a value into a stream after creation."